src/packet/buffer_rx.rs

```diff
@@ -167,80 +174,45 @@ impl DepacketizingBuffer {
             }
         };
 
-        let contiguous = self.contiguous(start, stop, &dep);
-
-        let is_more_than_hold_back = self.segments.len() >= self.hold_back;
+        // If we have contiguity of seq numbers we emit right away,
+        // Otherwise, we wait for retransmissions up to `hold_back` frames
+        // and re-evaluate contiguity based on codec specific information
 
-        let wait_for_contiguity = !contiguous && !is_more_than_hold_back;
+        let more_than_hold_back = self.segments.len() >= self.hold_back;
+        let contiguous_seq = self.is_following_last(start);
+        let wait_for_contiguity = !contiguous_seq && !more_than_hold_back;
```

Comment:
The amount of changes here seem a bit "not invented here" – unnecessary name change etc.

Reply:
Anything specific I can address ?